### PR TITLE
Remove route builder

### LINKED ISSUE #2870: Add access to Env in error handler
Hi!

Would it be possible to add support for accessing Env in e.g. 
Routes.handleErrorRequestCauseZIO(f: (Request, Cause[Err]) => ZIO[Any, Nothing, Response]): Route[Env, Nothing]

so it becomes 

Routes.handleErrorRequestCauseZIO(f: (Request, Cause[Err]) => ZIO[Env, Nothing, Response]): Route[Env, Nothing]? 
It would be great to be able to get logged in user, correlation id etc when generating the error message.

Thanks!

### PR DESCRIPTION
**Pull request by 987Nabil**
fixes #2870
/claim #2870

### CODE REVIEW

**Comment by jdegoes** [comment]
@987Nabil You referenced the wrong ticket?

**Comment by 987Nabil** [comment]
I removed the route builder, since it is not needed anymore. Also I enabled Env access for the zio effects that handle errors on `Route`/`Routes`

**Comment by 987Nabil** [comment]
@jdegoes it fixes both

File: zio-http/shared/src/main/scala/zio/http/Route.scala [hunk 1]
```diff
@@ -53,7 +53,9 @@ sealed trait Route[-Env, +Err] { self =>
   final def handleError(f: Err => Response)(implicit trace: Trace): Route[Env, Nothing] =
     self.handleErrorCauseZIO(c => ErrorResponseConfig.configRef.get.map(Response.fromCauseWith(c, _)(f)))
 
-  final def handleErrorZIO(f: Err => ZIO[Any, Nothing, Response])(implicit trace: Trace): Route[Env, Nothing] =
+  final def handleErrorZIO[Env1 <: Env](
```
Comment (987Nabil) [inline zio-http/shared/src/main/scala/zio/http/Route.scala:56]: Env access

File: zio-http/shared/src/main/scala/zio/http/Route.scala [hunk 2]
```diff
@@ -102,33 +102,44 @@ sealed trait Route[-Env, +Err] { self =>
    * can be used to convert a route that does not handle its errors into one
    * that does handle its errors.
    */
-  final def handleErrorCauseZIO(
-    f: Cause[Err] => ZIO[Any, Nothing, Response],
-  )(implicit trace: Trace): Route[Env, Nothing] =
+  final def handleErrorCauseZIO[Env1 <: Env](
```
Comment (987Nabil) [inline zio-http/shared/src/main/scala/zio/http/Route.scala:105]: env access

**Review by jdegoes — APPROVED** [review]

**Pull request merged**